src/app/assessment/assessment.component.ts

```diff
@@ -240,37 +277,59 @@ export class AssessmentComponent extends RouterEnter {
       });
     }
 
+    const params: SaveAnswersParams = {
+      assessment: assessment,
+      answers: answers,
+      action: this.action
+    };
+    if (this.submission) {
+      params.AssessmentSubmissionId = this.submission.id;
```

Comment:
As I mentioned in assessment.service, we don't need to create it as an interface. 
Here we can just initialise `submissionId` as `null`, if there is a submission, pass the actual submission id